### DISCUSSION TITLE
Allow connections from other hosts

### DISCUSSION THREAD

**Discussion by lesensei** [nabaztag2018/pynab#410] [Q&A] [answered]
I just brought my Nabaztag back to life thanks to Olivier and this here repository :-) This is amazing and anyone involved should get a huge "thank you" !
Since I'm a home automation enthusiast, I'd love to be able to connect my Node-Red instance to nabd. For reasons I think many here will understand, I prefer to stay away from cloud services like IFTTT.
Now, this shouldn't be a problem except I can't find how to allow non-localhost connections to nabd. I've tried modifying nabd.socket and nabservice.py to have "0.0.0.0" instead of "127.0.0.1", but that doesn't seem to do the trick.
Can anyone help with that ?
Thanks a lot !

**Comment by f-laurens** [comment, marked as answer]
> Now, this shouldn't be a problem except I can't find how to allow non-localhost connections to nabd. 

You might want to have a look at my [pynab_cli](https://github.com/f-laurens/pynab_cli) package, whose `pynab` command has options to handle this.

**Reply by lesensei** [reply]
Thank you. Looking at the source, I simply didn't edit the right files (I guess it was too late for my brain to function correctly). Thanks for this !

**Reply by f-laurens** [reply]
> Looking at the source, I simply didn't edit the right files 

Please note also that the `/lib/systemd/system/nabd.*` files are regenerated whenever Pynab is upgraded/re-installed.
So you you will need to reapply your change after an upgrade.

**Reply by gremlin-58** [reply]
.. should be fantastic to make an API via Django REST Framework and implement Token Authentication ..
